### PR TITLE
`improper_ctypes`: guard against accidental change to `Unique<T>`

### PR DESCRIPTION
**Pull request by Centril**
r? @eddyb

### CODE REVIEW

File: src/test/ui/lint/lint-ctypes-enum.rs [hunk 1]
```diff
@@ -44,6 +45,7 @@ extern {
    fn option_ref(x: Option<&'static u8>);
    fn option_fn(x: Option<extern "C" fn()>);
    fn nonnull(x: Option<std::ptr::NonNull<u8>>);
+   fn unqique(x: Option<std::ptr::Unique<u8>>); //~ ERROR enum has no representation hint
```
Comment (rkruppe) [inline src/test/ui/lint/lint-ctypes-enum.rs]:
```suggestion
   fn unique(x: Option<std::ptr::Unique<u8>>); //~ ERROR enum has no representation hint
```

**Review by hanna-kruppe — APPROVED** [review]
r=me after fixing the typo

File: src/test/ui/lint/lint-ctypes-enum.stderr [hunk 2]
```diff
@@ -1,82 +1,90 @@
...
-  --> $DIR/lint-ctypes-enum.rs:10:1
+  --> $DIR/lint-ctypes-enum.rs:11:1
    |
 LL | enum T { E, F, G }
    | ^^^^^^^^^^^^^^^^^^
 
+error: `extern` block uses type `std::option::Option<std::ptr::Unique<u8>>` which is not FFI-safe: enum has no representation hint
+  --> $DIR/lint-ctypes-enum.rs:48:18
+   |
+LL |    fn unqique(x: Option<std::ptr::Unique<u8>>);
```
Comment (Centril) [inline src/test/ui/lint/lint-ctypes-enum.stderr]:
```suggestion
LL |    fn unique(x: Option<std::ptr::Unique<u8>>);
```

**Comment by Centril** [comment]
Argh; messed the manual blessing up... Will fix in a bit. @bors r-

**Pull request merged**